src/ansys/fluent/core/utils/file_transfer_service.py

```diff
@@ -232,15 +213,37 @@ def upload(
         FileNotFoundError
             If a file does not exist.
         """
-        files = _get_files(file_name, self.host_mount_path)
+        files = _get_files(file_name)
         if self.client:
             for file in files:
-                is_file_on_remote = self.file_exists_on_remote(os.path.basename(file))
-                if is_file_on_remote:
+                full_file_name, is_file_on_remote = self.file_exists_on_remote(
+                    os.path.basename(file)
+                )
+                if is_file_on_remote and full_file_name.samefile(file):
                     warnings.warn(
                         f"\n{file} with the same name exists at the remote location.\n",
                         UserWarning,
                     )
+                    return
```

Comment:
If local and remote files are same then don't upload local file.

Reply:
I think we should use this  logic in download as well.

Reply:
I guess this condition corresponds to if the local and remote path point to the same physical location.

Reply:
Yes, right.